packages/framework-provider-azure/src/library/events-store-adapter.ts

```diff
@@ -0,0 +1,140 @@
+      if (batchResponse?.code !== 200) {
+        logger.error(batchResponse)
+        throw new Error(`Error ${batchResponse.substatus} storing events`)
+      }
+      const result = batchResponse.result
+      logger.debug(`EventEnvelopes with ${partitionKey} stored: ${result}`)
+    }
+  }
+
+  return envelopesWithCreatedAt
+}
+
+function toEventEnvelopes(events: Array<NonPersistedEventEnvelope>): Array<EventEnvelope> {
+  return events.map((eventEnvelope: NonPersistedEventEnvelope) => ({
+    ...eventEnvelope,
+    createdAt: new Date().toISOString(),
+  }))
+}
+
+function toInputOperations(
+  eventEnvelopesChunkWithCreatedAt: Array<EventEnvelope>,
+  config: BoosterConfig
+): Array<CreateOperationInput> {
+  return eventEnvelopesChunkWithCreatedAt.map((eventEnvelopeWithCreatedAt) => {
+    const body: JSONObject = storableResource(config, eventEnvelopeWithCreatedAt)
+    return {
+      operationType: 'Create',
+      resourceBody: body,
+    }
+  })
+}
+
+async function batchEvents(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  inputOperations: Array<CreateOperationInput>,
+  partitionKey: string
+): Promise<Response<OperationResponse[]>> {
+  const logger = getLogger(config, 'store-events-adapter#batchEvents')
+  try {
+    logger.debug('Storing EventEnvelopes with inputOperations:', inputOperations)
+    return await cosmosDb
+      .database(config.resourceNames.applicationStack)
+      .container(config.resourceNames.eventsStore)
+      .items.batch(inputOperations, partitionKey)
+  } catch (e) {
+    logger.error('Unexpected error storing events', e)
+    throw e
+  }
+}
+
+function groupByPartitionKey(
+  eventsPerPartitionKey: NonPersistedEventEnvelopePerPartitionKey,
+  event: NonPersistedEventEnvelope
+): NonPersistedEventEnvelopePerPartitionKey {
+  const partitionKey: string = partitionKeyForEvent(event.entityTypeName, event.entityID)
+  if (!eventsPerPartitionKey[partitionKey]) {
+    eventsPerPartitionKey[partitionKey] = []
+  }
+  eventsPerPartitionKey[partitionKey].push(event)
+  return eventsPerPartitionKey
+}
+
+/**
+ * Batch method expects a JSONObject. JSONObject valid types are: boolean | number | string | null | JSONArray | JSONObject
+ * This method tries to build a JSONObject from an EventEnvelope.
+ * It could fail if value contains a circular reference or a BigInt value is encountered.
+ * @param config
+ * @param eventEnvelope
+ */
+function storableResource(config: BoosterConfig, eventEnvelope: EventEnvelope): JSONObject {
+  const logger = getLogger(config, 'store-events-adapter#storableResource')
+  try {
+    const partitionKey = partitionKeyForEvent(eventEnvelope.entityTypeName, eventEnvelope.entityID)
+    const eventEnvelopeWithEntityID = {
+      ...eventEnvelope,
+      [eventsStoreAttributes.partitionKey]: partitionKey,
+      [eventsStoreAttributes.sortKey]: eventEnvelope.createdAt,
+    }
+    return JSON.parse(JSON.stringify(eventEnvelopeWithEntityID))
```

Comment:
I guess this is done to verify the circular dependencies and the "BigInt" kind of value stated in the above comment, but it seems a bit hacky (and time consuming... as this will be used to store batches of hundred of events).

My question is: is this really needed? a Javascript object like the `eventEnvelope` is already a JSONObject. 
Also, if cosmos will fail when cyclic dependencies or BibInts are present, then we can let it fail.

In brief: could we just return here `eventEnvelopeWithEntityID`?

Reply:
@gonzalogarciajaubert what do you think about @alvaroloes's suggestion?

Reply:
@alvaroloes in commit f9793a60 this return was simplified following your suggestion. I spoke to @gonzalogarciajaubert and it appears there shouldn't be an issue with this (very unlikely to get a BigInt or an object with circular dependencies)